satpy/readers/atms_l1b_nc.py

```diff
@@ -0,0 +1,153 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2022 Satpy developers
+#
+# satpy is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# satpy is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with satpy.  If not, see <http://www.gnu.org/licenses/>.
+"""Advanced Technology Microwave Sounder (ATMS) Level 1B product reader.
+
+The format is explained in the `ATMS L1B Product User Guide`_
+
+.. _`ATMS L1B Product User Guide`:
+  https://docserver.gesdisc.eosdis.nasa.gov/public/project/Sounder/ATMS_V3_L1B_Product_User_Guide.pdf
+
+"""
+
+import logging
+from datetime import datetime
+
+from satpy.readers.netcdf_utils import NetCDF4FileHandler
+
+logger = logging.getLogger(__name__)
+
+DATE_FMT = '%Y-%m-%dT%H:%M:%SZ'
+ATMS_CHANNEL_NAME_TO_INDEX = {
+    "1": 0,
+    "2": 1,
+    "3": 2,
+    "4": 3,
+    "5": 4,
+    "6": 5,
+    "7": 6,
+    "8": 7,
+    "9": 8,
+    "10": 9,
+    "11": 10,
+    "12": 11,
+    "13": 12,
+    "14": 13,
+    "15": 14,
+    "16": 15,
+    "17": 16,
+    "18": 17,
+    "19": 18,
+    "20": 19,
+    "21": 20,
+    "22": 21,
+}
```

Comment:
I'm not sure I understand why this is needed. Isn't this just `int(channel_name) - 1`? It might be cleaner and less code overall if in the data loading you did something like:

```
try:
    channel_index = int(name) - 1
    return self.antenna _temperature[:, :, channel_index]
except TypeError:
    return self[name]
```

then you don't have to have this dictionary or the overridden `__init__`.

Reply:
Thanks, your suggestion is much cleaner. Fixed it.